### PR TITLE
 Fixed bug with RedundantBraces for string interpolation

### PR DESCRIPTION
**Pull request by poslegm**
Fixes (#1420)
Fixes (#1227)

Also includes tests from #1421

### CODE REVIEW

**Comment by poslegm** [comment]
Hmm, I don't understand what's happening with commit difference :thinking: 

Has the master been rewritten? Should I create new fork or is it ok?

File: scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/RedundantBraces.scala [hunk 1]
```diff
@@ -32,9 +32,16 @@ case object RedundantBraces extends Rewrite {
     def isIdentifierAtStart(value: String) =
       value.nonEmpty && (Character.isLetterOrDigit(value.head) || value.head == '_')
 
+    // hack for literal identifiers detection (`type`, `val`, ...)
+    def isLiteralIdentifier(arg: Term.Name): Boolean =
```
Comment (poslegm) [inline scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/RedundantBraces.scala]: Is there better solution for it?

Comment (tanishiking) [reply scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/RedundantBraces.scala]: How about this?

```scala
arg.syntax.startsWith("`") && arg.syntax.endsWith("`")
```

File: scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/RedundantBraces.scala [hunk 2]
```diff
@@ -32,9 +32,16 @@ case object RedundantBraces extends Rewrite {
...
+      (arg.pos.end - arg.pos.start) - arg.value.length() == 2
+
+    def shouldTermBeEscaped(arg: Term.Name): Boolean =
```
Comment (tanishiking) [inline scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/RedundantBraces.scala]: Could we add a comment that refers to https://github.com/scalameta/scalafmt/issues/1420 ?
This solution is a bit hacky and worth explaining why this condition is required :)

File: scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/RedundantBraces.scala (hunk 1 above)
Comment (poslegm) [reply scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/RedundantBraces.scala]: Oh, I didn't know about `Syntax` typeclass. Thank you very much!

File: scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/RedundantBraces.scala (hunk 2 above)
Comment (poslegm) [reply scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/RedundantBraces.scala]: done

**Pull request merged**